From c982d674a96716e8e955784a631bc37d59242c19 Mon Sep 17 00:00:00 2001
From: Jernej Skrabec <jernej.skrabec@gmail.com>
Date: Fri, 22 May 2026 16:51:14 +0000
Subject: [PATCH 30/33] drm/sun4i: Add A523 dw-hdmi compatible

A523/A527/T527 reuses the Synopsys DesignWare HDMI controller. Add a
new compatible with quirks matching H6 (mode_valid up to 594 MHz,
DRM-style infoframe support).

Signed-off-by: Jernej Skrabec <jernej.skrabec@gmail.com>
---
 drivers/gpu/drm/sun4i/sun8i_dw_hdmi.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/gpu/drm/sun4i/sun8i_dw_hdmi.c b/drivers/gpu/drm/sun4i/sun8i_dw_hdmi.c
index 96532709c..e944643a2 100644
--- a/drivers/gpu/drm/sun4i/sun8i_dw_hdmi.c
+++ b/drivers/gpu/drm/sun4i/sun8i_dw_hdmi.c
@@ -249,6 +249,11 @@ static const struct sun8i_dw_hdmi_quirks sun50i_h6_quirks = {
 	.use_drm_infoframe = true,
 };
 
+static const struct sun8i_dw_hdmi_quirks sun55i_a523_quirks = {
+	.mode_valid = sun8i_dw_hdmi_mode_valid_h6,
+	.use_drm_infoframe = true,
+};
+
 static const struct of_device_id sun8i_dw_hdmi_dt_ids[] = {
 	{
 		.compatible = "allwinner,sun8i-a83t-dw-hdmi",
@@ -258,6 +263,10 @@ static const struct of_device_id sun8i_dw_hdmi_dt_ids[] = {
 		.compatible = "allwinner,sun50i-h6-dw-hdmi",
 		.data = &sun50i_h6_quirks,
 	},
+	{
+		.compatible = "allwinner,sun55i-a523-dw-hdmi",
+		.data = &sun55i_a523_quirks,
+	},
 	{ /* sentinel */ },
 };
 MODULE_DEVICE_TABLE(of, sun8i_dw_hdmi_dt_ids);
-- 
2.53.0

